### PR TITLE
Clean-up explore tests

### PR DESCRIPTION
**Pull request by poornachandra**

### CODE REVIEW

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-DUT1283

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceTestsSuite.java [hunk 1]
```diff
@@ -31,12 +33,17 @@
   HiveExploreServiceTestRun.class,
   WritableDatasetTestRun.class,
   HiveExploreObjectMappedTableTestRun.class,
-  HiveExploreServiceStopTestRun.class
```
Comment (albertshau) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceTestsSuite.java]: why remove?

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceTestsSuite.java [hunk 2]
```diff
@@ -31,12 +33,17 @@
...
+  HiveExploreServiceFileSetTest.class
```
Comment (albertshau) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceTestsSuite.java]: should rename to HiveExploreServiceFileSetTestRun so that maven will not run it both in the suite and outside the suite

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/BaseHiveExploreServiceTest.java [hunk 3]
```diff
@@ -367,6 +373,16 @@ protected void configure() {
                     .build(StoreFactory.class)
           );
           bind(StreamHttpService.class).in(Scopes.SINGLETON);
+
+          install(Modules.override(new DataFabricModules().getInMemoryModules()).with(new AbstractModule() {
```
Comment (albertshau) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/BaseHiveExploreServiceTest.java]: why is this needed?

**Comment by albertshau** [comment]
couple questions and one comment

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceTestsSuite.java (hunk 1 above)
Comment (poornachandra) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceTestsSuite.java]: Since `HiveExploreServiceStopTestRun` stops the service, it is a good idea not to run it as part of the test suite.

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/BaseHiveExploreServiceTest.java (hunk 3 above)
Comment (poornachandra) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/BaseHiveExploreServiceTest.java]: I changed the way transaction was asserted in `HiveExploreServiceStopTestRun`, so that we don't have conditional logic in `BaseHiveExploreServiceTest` code. Now all services get stopped, and transaction state is asserted using transaction snapshot.

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceTestsSuite.java (hunk 2 above)
Comment (poornachandra) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceTestsSuite.java]: The JUnit docs don't say anything about excluding Test files that are part of TestSuite from outside the suite. But I see that our config is setup that way. I'll fix the class names.

**Comment by poornachandra** [comment]
@albertshau Addressed review comments and did some more refactoring. Please take a look.

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/BaseHiveExploreServiceTest.java [hunk 4]
```diff
@@ -138,24 +144,23 @@
 
   protected static Injector injector;
 
-  @ClassRule
-  public static TemporaryFolder tmpFolder = new TemporaryFolder();
-
-  protected static void initialize() throws Exception {
-    initialize(CConfiguration.create());
+  protected static void initialize(TemporaryFolder tmpFolder) throws Exception {
```
Comment (albertshau) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/BaseHiveExploreServiceTest.java]: how come it has to be passed in? isn't it get a new folder each time initialize is called?

Comment (poornachandra) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/BaseHiveExploreServiceTest.java]: I'm now passing in `TemporaryFolder` in initialize, otherwise when we call `intitalize` from `BaseHiveExploreServiceTest` the `@ClassRule` may not be initiailzed.

**Comment by albertshau** [comment]
:+1:

**Comment by poornachandra** [comment]
Build is green http://builds.cask.co/browse/CDAP-DUT1283.
